docs/CODING_STYLE.rst

```diff
@@ -67,32 +67,219 @@ Coding guide
 
 Comments
 --------
+Before every function prototype in ``.h`` files, include a Doxygen-formatted comment
+like this:
 
-Before every function have in ``.h`` files a comment like this:
+.. code-block:: c
 
-.. code:: c
+    /**
+     * Brief description
+     *
+     * @return    short description of return value
+     */
+    example_type_t * lv_example(void);
 
-   /**
-    * Return with the screen of an object
-    * @param obj pointer to an object
-    * @return pointer to a screen
-    */
-   lv_obj_t * lv_obj_get_screen(lv_obj_t * obj);
-
-Always use ``/*Something*/`` format and NOT ``//Something``
+Another example with arguments and more detail.
 
-Write readable code to avoid descriptive comments like:
-``x++; /*Add 1 to x*/``. The code should show clearly what you are
-doing.
+.. code-block:: c
 
-You should write **why** have you done this:
-``x++; /*Because of closing '\0' of the string*/``
+    /**
+     * Brief description
+     *
+     * Additional detail here in subsequent paragraphs.  Function accomplishes these results:
+     * - description of result 1
+     * - description of result 2
+     *
+     * @param  disp       object created by lv_x11_window_create()
+     * @param  mouse_img  optional image description for mouse cursor
+     *                        (NULL for no/invisible mouse cursor)
+     *
+     * @note  Include any subtle points an API user or maintainer would need to know.
```

Comment:
Shall we show `code` and `example` too?

Reply:
> Shall we show `code` and `example` too?

I'm not sure I am getting the concept you trying to convey here.  Can you clarify?  Are you referring to making the example more complete so it shows we encourage placing code elements in single back-quotes?

Reply:
I just meant that we support more Doxygen tags (`@....`) and we can show them in a single complete example.